extensions/amp-a4a/0.1/test/test-amp-a4a.js

```diff
@@ -124,6 +139,198 @@ describe('amp-a4a', () => {
     return element;
   }
 
+  describe('#extractCreativeAndSignature', () => {
+    it('should extract creative and signature if they exist', () => {
+     viewerWhenVisibleMock.onFirstCall().returns(Promise.resolve());
+      xhrMock.withArgs('https://test.location.org/ad/012345?args', {
+        mode: 'cors',
+        method: 'GET',
+        credentials: 'include',
+        requireAmpResponseSourceOrigin: true,
+      }).onFirstCall().returns(Promise.resolve(mockResponse));
+      return createAdTestingIframePromise().then(fixture => {
+        const doc = fixture.doc;
+        const a4aElement = createA4aElement(doc);
+        a4aElement.setAttribute('width', 200);
+        a4aElement.setAttribute('height', 50);
+        a4aElement.setAttribute('type', 'adsense');
+        const a4a = new MockA4AImpl(a4aElement);
+	const extractCreativeAndSignatureSpy = sandbox.spy(
+            a4a, 'extractCreativeAndSignature');
+        doc.body.appendChild(a4aElement);
+        a4a.onLayoutMeasure();
+        expect(a4a.adPromise_).to.be.instanceof(Promise);
+        return a4a.adPromise_.then(() => {
+          // Force vsync system to run all queued tasks, so that DOM mutations
+          // are actually completed before testing.
+          a4a.vsync_.runScheduledTasks_();
+	  expect(extractCreativeAndSignatureSpy.calledOnce,
+		 'extractCreativeAndSignatureSpy called exactly once').to.be.true;
```

Comment:
Can't really test that though, as that is something implemented by each ad network. If I test what I get back, I'm just testing that the MockA4AImpl does something correctly. Can only test that it gets called properly, no?